src/views/course_pairings.vue

```diff
@@ -0,0 +1,215 @@
+
+        <p>Specify Course: </p>
+        <select v-model="specify_course">
+            <option>All Courses</option>
+            <option v-for="course in this.courses_list">{{course}}</option>
+        </select>
+
+        <p>Specify Term: </p>
+        <select v-model="specify_term">
+            <option>All Terms</option>
+            <option v-for="term in this.list_terms">{{term}}</option>
+        </select>
+
+        <p>Group By</p>
+        <select v-model="column" options="this.column">
+            <option>Instructor</option>
+            <option>TA</option>
+        </select>
+    </div>
+
+    <div id="table"></div>
+</div>
+
+</template>
+
+<script>
+import HotTable from "@handsontable/vue";
+import Handsontable from "handsontable";
+import generate_query_string from "./components/generate_query_string";
+
+export default {
+    data: function() {
+        return {
+            table: null,
+            specify_term: "All Terms",
+            column: "Instructor",
+            root: "table-format",
+            specify_course: "All Courses",
+            user_associations: [],
+            courses_list: [],
+            list_terms: [],
+            tableSettings: {
+                data: [],
+                stretchH: "all",
+                colHeaders: []
+            }
+        };
+    },
+    /**
+     * Function is called when page is created to get course pairings from API.
+     * Data is used to populate the spreadsheet and options for select
+     */
+    created: function() {
+        this.getData();
+    },
+
+    watch: {
+        /**
+         * Function is called when specify_course is modified. The data is used
+         * to populate the page with the specified course.
+         */
+        specify_course: function() {
+            this.getData();
+        },
+
+        /**
+         * Function is called when specify_term is modified. The data is used
+         * to populate the page with the specified term.
+         */
+        specify_term: function() {
+            this.getData();
+        },
+
+        /**
+         * Function is called when column is modified. The data is used
+         * to populate the page with the specified column.
+         */
+        column: function() {
+            this.getData();
+        }
+    },
+    methods: {
+        /**
+         * Function gets course pairing data from API for the specified
+         * course, term and column and calls parseData to parse retuned data.
+         */
+        getData: function() {
+            // Converting to parameter enums
+            var column_name = this.column === "TA" ? "ta" : "instructor";
+
+            var course_where = "";
+            var term_where = "";
+            if (this.specify_term != "All Terms") {
+                term_where = `${this.specify_term}`;
+            }
+            if (this.specify_course != "All Courses") {
+                course_where = `${this.specify_course}`;
+            }
+
+            let url = `{
```

Comment:
This is not right. Just define an object straight up. Not a string, which gets turned into an object, which gets reparsed again.